### PR TITLE
Add get_data_path to resource

### PR DESCRIPTION
**Pull request by flowerbug**
For linux and posix type systems that follow the following specification there is a separate data directory for an application under XDG_DATA_HOME or if the user or system admin has not set the environment variable the location is set to "./local/share/Application-Name"  the added function returns that name to the caller.  For non-linux system platforms I used the same directories returned by get_settings_path as I do not own or know those platforms, but I figure that anyone who uses this path will know what they need it for.

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

### CODE REVIEW

**Comment by flowerbug** [comment]
> Thanks for the PR! This was needed.

You're welcome!

I kept things very simple with this PR, but my actual hope was to improve it too eventually.  There are two issues.

One is that the script requires an input name at all.  There should be an option to not specify a name at all and have the function just return the name of the program.

The other is that the name of the program might have an extension like .py on it and the normal usage in the directories is to not have that extension on there at all.  So every time the user is calling the function they are not only having to get the name but they also have to take it apart before calling the function.  As a helper function and since sys is already being imported why not do this for the user too?  :)

**Review by benmoran56 — CHANGES_REQUESTED** [review]
Do you mind making that small change on line 235? If it's difficult, I can do it on my end.

File: pyglet/resource.py [hunk 1]
```diff
@@ -196,6 +196,47 @@ def get_settings_path(name):
...
+            return os.path.join(os.environ['APPDATA'], name)
+        else:
+            return os.path.expanduser('~/%s' % name)
+    elif pyglet.compat_platform == 'darwin':
+        return os.path.expanduser('~/Library/Application Support/%s' % name)
+    elif pyglet.compat_platform.startswith('linux'):
+        if 'XDG_DATA_HOME' in os.environ:
+            return os.path.join(os.environ['XDG_DATA_HOME'], name)
+        else:
+            return os.path.expanduser('~/.config/share/%s' % name)
```
[the remark this answers is not in the document]
Comment (flowerbug) [reply pyglet/resource.py:235]: yes, that is good, i had it right in the doc above.  i would have caught it eventually as i will be using it.

**Review by benmoran56 — APPROVED** [review]
Looks good!

**Pull request merged**